src/chain/shared_state.rs

```diff
@@ -104,116 +96,30 @@ impl SharedState {
         }
     }
 
-    pub fn update_with_genesis_related_info(
-        &mut self,
-        related_info: &[u8],
-        log_ident: &LogIdent,
-    ) -> Result<(), RoutingError> {
-        update_with_genesis_related_info_check_same(
-            log_ident,
-            "handled_genesis_event",
-            &self.handled_genesis_event,
-            &false,
-        );
-        self.handled_genesis_event = true;
-
-        if related_info.is_empty() {
-            return Ok(());
+    pub fn update(&mut self, new: Option<Self>, log_ident: &LogIdent) {
+        if self.handled_genesis_event {
+            log_or_panic!(
```

Comment:
Is this an example of a straigh log message. I am not sure we would want to panic in a test here?

Reply:
This exists in fleming too. We can change it (as per the ongoing discussion about error handling), but best to do it in a separate PR focused on just that.